File: Lecture07_Navigation_UI/AfterWorkshop_Java/app/src/main/java/com/lavima/afterworkshop_java/MainActivity.java
package com.lavima.afterworkshop_java;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationView;

import java.util.HashSet;

public class MainActivity extends AppCompatActivity {

    private NavController controller;
    private DrawerLayout drawerLayout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Find navigation controller in the navigation host
        controller = Navigation.findNavController(this, R.id.fragment);

        // Find drawer layout
        drawerLayout = findViewById(R.id.drawer_layout);

        // The following activates the bottom navigation menu. The menu item ids match destination ids so no explicit
        // handling is necessary

        BottomNavigationView bottomNavigation = findViewById(R.id.bottom_navigation);

        // Connect with navigation controller for automatic navigation
        NavigationUI.setupWithNavController(bottomNavigation, controller);
        // Simple explicit navigation
        //bottomNavigation.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
        //    @Override
        //    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
        //        controller.navigate(menuItem.getItemId());
        //        return true;
        //    }
        //});

        // This will enable us to click the toolbar/appbar to open drawer

        Toolbar toolbar = findViewById(R.id.toolbar);
        // This will make sure that we don't get a back/up-arrow on the top level destinations
        AppBarConfiguration configuration = new AppBarConfiguration.Builder(R.id.homeFragment, R.id.searchFragment, R.id.collectionFragment)
                .setOpenableLayout(drawerLayout).build();
        NavigationUI.setupWithNavController(toolbar, controller, configuration);


        // The following activates the navigation drawer.

        NavigationView drawerView = findViewById(R.id.navigation_view);
        // Connect with navigation controller for automatic navigation (menu ids must match destination ids)
        NavigationUI.setupWithNavController(drawerView, controller);
        // Simple explicit navigation
        drawerView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                controller.navigate(menuItem.getItemId());
                drawerLayout.closeDrawers();
                return true;
            }
        });

    }
}

File: Lecture07_Navigation_UI/After_Kotlin_02_BottomNav/app/src/main/java/no/hiof/larseknu/playingwithbottomnavigation/MainActivity.kt
package no.hiof.larseknu.playingwithbottomnavigation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Gets a reference to the navController
        var navController = findNavController(R.id.navHostFragment)

        // Sets up bottom navigation
        setupBottomNavMenu(navController)
    }

    private fun setupBottomNavMenu(navController: NavController) {
        // Gets a reference to the bottom navigation view
        val bottomNav = findViewById<BottomNavigationView>(R.id.bottomNavigationView)
        // Sets it up with the navigationcontroller (and navigation graph)
        bottomNav?.setupWithNavController(navController)

        // If the id in the menu and navigation graph are the same, this is done automagically
        /*bottomNav.setOnNavigationItemSelectedListener { menuItem ->

            when (menuItem.itemId) {
                R.id.home_destination -> navController.navigate(R.id.home_dest)
                R.id.search_destination -> navController.navigate(R.id.search_dest)
                R.id.collection_destination -> navController.navigate(R.id.collection_dest)
            }

            true
        }*/
    }
}
